components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -92,7 +92,17 @@ export default function UsageBasedBillingConfig({ attributionId }: Props) {
             setPendingStripeSubscription(pendingSubscription);
             window.localStorage.setItem(localStorageKey, JSON.stringify(pendingSubscription));
             try {
-                await getGitpodService().server.subscribeToStripe(attributionId, setupIntentId);
+                // Pick a good initial value for the Stripe usage limit (1000 * team_size)
+                // FIXME: Should we ask the customer to confirm or edit this default limit?
+                let limit = 1000;
+                const attrId = AttributionId.parse(attributionId);
+                if (attrId?.kind === "team") {
+                    const members = await getGitpodService().server.getTeamMembers(attrId.teamId);
+                    limit = 1000 * members.length;
+                }
+                await getGitpodService().server.subscribeToStripe(attributionId, setupIntentId, limit);
```

Comment:
This would allow me to set the usage limit to an arbitrary value because we just pass that into the SetCostCenter call. Where do we validate this?

Reply:
While there is some front-end validation, I'm relying on the backend (Cost Center code) to be the final validator. The goal here is to eventually be able to ask users what sort of limit they would like when they sign up, but I think the backend should be the main authority on what is or isn't acceptable.

Reply:
Many thanks for implementing the back-end validation in https://github.com/gitpod-io/gitpod/pull/13798! 👍